extensions/amp-social-share/1.0/social-share.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as CSS from './social-share.css';
+import * as Preact from '../../../src/preact';
+import {Keys} from '../../../src/utils/key-codes';
+import {SocialShareIcon} from '../../../third_party/optimized-svg-icons/social-share-svgs';
+import {addParamsToUrl, parseQueryString} from '../../../src/url';
+import {dict} from '../../../src/utils/object';
+import {getSocialConfig} from './amp-social-share-config';
+import {isObject} from '../../../src/types';
+import {openWindowDialog} from '../../../src/dom';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+const DEFAULT_WIDTH = 60;
+const DEFAULT_HEIGHT = 44;
+const NAME = 'SocialShare';
+/**
+ * @param {!JsonObject} props
+ * @return {PreactDef.Renderable}
+ */
+export function SocialShare(props) {
+  useResourcesNotify();
+  const {typeConfig, baseEndpoint, checkedWidth, checkedHeight} = checkProps(
+    props,
+    NAME
+  );
+  const finalEndpoint = createEndpoint(typeConfig, baseEndpoint, props);
+
+  const type = props['type'].toUpperCase();
+  const baseStyle = CSS.BASE_STYLE;
+  const backgroundStyle = CSS[type] || CSS.DEFAULT;
+  const size = {
+    width: checkedWidth,
+    height: checkedHeight,
+  };
+  return (
+    <div
+      role="button"
+      tabindex={props['tabIndex'] || '0'}
+      onKeyDown={(e) => handleKeyPress(e, finalEndpoint)}
+      onClick={() => handleActivation(finalEndpoint)}
+      style={{...size, ...props['style']}}
+      {...props}
+    >
+      <SocialShareIcon
+        style={{...backgroundStyle, ...baseStyle, ...size}}
+        type={type}
+      />
+    </div>
+  );
+}
+
+/**
+ * @param {!JsonObject} props
+ * @param {?string} name
+ * @return {struct}
```

Comment:
This will look like this:

```
@return {{
  typeConfig: !JsonObject,
  baseEndpoint: string,
  checkedWidth: number,
  checkedHeight: number,
}}
```

Reply:
Thanks, I have updated the struct definition!